### PR TITLE
Improvised Weapons Update [Read Changelog]

### PR DESCRIPTION
**Pull request by KathrinBailey**
## About The Pull Request
**TL;DR: Adds six weapons to the game, two are improvised weapons. Improvised weapon crafting has harder in general and the improvised weapon meta has been tweaked to make the shotgun not completely dominant. Also new sprites for the improvised weapons and new ammo types.**

• Adds the Type-19, Type-20 and Type-21 Encased Energy Director. 
   • Type 21 is functionally identical to the HoS laser.
   • Type 20 is functionally identical to a laser gun.
   • Type 19 is worse than a laser gun, it fires three round bursts that add up to 18 damage with 
     slightly lower energy cost. This is for events and will be utilised in a future PR. Does 5 stamina 
     damage, 15 total as well. Part of the suppression gimmick.
• Adds the Sumpter Bullpup rifle. A 5.56 semi/2rnd-burst rifle similar to the WT550, but with more damage and less ammo per magazine.
**The Sumper and EED are currently only available in-game via admin-spawn. Future PRs will utilise these for events and potentially a PR to rework cargo crates.**
• Adds two new improvised weapons: the improvised pistol and the improvised laser. Both of which are worse than the improvised shotgun and rifle, but they fill niche roles. The pistol has horrible accuracy, at a range of four tiles you'll start missing. It has a high magazine size (8) and low damage (13). It is easily concealable and very small, it can fit inside pockets. You must craft an empty pistol magazine and print the ammo from an autolathe and put it in the magazine manually (stops overwatch Reaper reloading). The laser has a low capacity (5) and low damage (10). It can be duel-wielded and recharged. It can also be upgraded by going through a 6-part intensive process using the cargo glassworking crate and tier 4 parts to go to 15 damage and a slightly higher capacity.
• Improvised Rifle has a new sprite.
• Improvised Shotgun has a new sprite.
• New weapon parts:
   • Barrels: Rifle, Shotgun & Pistol
   • Receivers: Rifle, Shotgun, Pistol & Laser Assembly
   • Misc: Trigger Assembly, Makeshift Lens, Wooden Grip & Wooden Body.
• 32 ACP ammo type introduced for the improvised pistol. This may be utilised for a fast-firing SMG later.
• 20 Gauge ammo type introduced for the improvised shotgun. This may be utilised for more shotguns in the future.
• M-90GL changed to fit a new niche as a PDW. The M-90 used to just be a C-20r with a grenade launcher. It now does less damage but with a 50rnd magazine, so it's more or less a real life P90. TC cost has been lowered to 12 from 18 to reflect this. The 40mm is still powerful. This change was mostly so I could use 5.56 for something else. It's still a powerful gun, it's just not a direct upgrade of the C-20r.
• A hand saw! With multi-coloured handles! It also fits on toolbelts! _Nostalgic sawing noises_
• Adds loot spawners for weapon parts, so you can have bits of them spawn in maint.

## Why It's Good For The Game
Improvised weapons have long been obscenely powerful for their almost free crafting costs. This PR hopes to address this by making improvised weapons harder to mass-produce. You can still get the weapons, they aren't locked behind departments or anything. It's just a bit harder and takes a bit longer and requires tools (the tools aren't super hard to get and are in unhacked autolathes too).

**Here are the goals I set out on when making this PR.**
   • Make improvised weapons harder to craft, but not obnoxiously hard or impossible to craft.
   • Shift the shotgun out of the prime spot in the meta, make it slightly worse, but not bad.
   • Add more variety to improvised weapons without making them  the new meta.
   • Make this change **fun** for all players.
The first three goals have definitely been met, I look forward to seeing if people find this to be  a fun change.

Improvised shotgun meta has been nerfed. Before it was a direct upgrade to the rifle, cheaper, the same damage, faster to fire, just as accurate, can do more damage point-blank with buckshot. Using 20 Gauge reels in the shotgun. Buckshot: 75 brute => 60 brute. Slugs: 60 brute => 50 brute. The rifle bullets deal 60 brute and is easier to craft than the shotgun (still harder than before this PR), so the rifle actually is worth being in the game now.

New improvised weapon variety fills niches. Want to conceal your weapon easily? Use the pistol. You won't be doing much damage, but you can hide it. Concerned about ammo? Use the laser. You can only get five shots off before you need to recharge it though Even then, the damage is even worse than the pistol. Somewhat concealable but the most expensive weapon to craft. Overall, the more variety in weapons should make it more fun. The more parts required to craft should give a sense of accomplishment when you finally get your shiny gun.

I haven't mentioned the Type 19/20/21 yet, so here it is.
People have liked the sprites for them, the Type 19 will be great for events. It's very scary to the crew, but it doesn't actually hurt them that much. It gives players the sense of being overwhelmed without actually overwhelming them, it's very suppressive and gives the player a sense of accomplishment for overcoming it. Type 20 is a laser gun, Type 21 is a HoS multiphase.

Sumpter may be introduced in a cargo crate rework, so far it's staying adminspawn for events. 

## Addressing Concerns
Alright, everyone is concerned about crafting costs. Some people may be worried that I'm nerfing the weapons into the ground, some people are concerned that I'm just going to make this frustrating and tedious. Here is the crafting menu so you can judge for yourself.
image
image
You need tools for these. Very few parts are available in the autolathe. The parts are still relatively easy to obtain, however, this is still a nerf. Everything is obtainable by an assistant. You can deconstruct most of reinforced walls to get bits of plasteel. Igniters are in the assistance vendors.
So, the new weapons. Are they comparable to the shotgun or rifle? _They're worse._

## Changelog
:cl: Kathrin Morrison / Floof Ball
add: Type-19, Type-20 and Type-21 EED.
add: Bullpup 5.56 rifle called 'Sumpter'
add: Improvised Pistol
add: Improvised Laser + Upgrade making use of the glassworking update.
add: New improvised weapon crafting parts
add: New weapon calibre: 20 gauge, used for the improvised shotgun.
add: A hand saw!
balance: Improvised weapon crafting takes more parts and more time. You can no longer mass-produce very powerful weapons in minutes.
balance: Improvised shotgun meta nerfed by moving improvised shotguns to 20 gauge, does slightly less damage. Still powerful, but not just a straight upgrade from the rifle in every way.
fix: fixed a few things
tweak: Syndicate P90 now uses 5.7x28mm instead of 5.56 like a normal P90. It does less damage and has a much bigger magazine, freeing up 5.56 for other stuff and making it not just "the C-20r with a grenade launcher". Lowered TC cost.
soundadd: New rifle sound for the Sumpter bullpup.
imageadd: New sprites for the improvised rifle, improvised shotgun, slung ishotgun + sawn ishotgun.
/:cl:

## Media
[CLICK HERE TO SEE THE WEAPON SPRITES](https://i.gyazo.com/5874c0aa85fbf72e75e4c100505e70fc.mp4)
^ ^ CLICK ^ ^ CLICK ^ ^ CLICK ^ ^ (Also fullscreen it)
**CLICK THE ABOVE LINK!!**
image
Sadly just a still image :( Click the above linkkkk!
image
image
image
image
image
image
image

There was going to be a sick-ass trailer for this PR but the people who volunteered for it beforehand were busy. :(

## Acknowledgements
My American friend who gave me the good idea of changing the shotgun ammo type to nerf its damage. Same for using 32ACP. I just went to my American and got some gun knowledge.
Feasel / Riastlin for discussing the changes and discussing balance with me.
Trillby for discussing the changes with me.
Justin / MrJWhit for discussing the changes with me.

### CODE REVIEW

File: code/modules/projectiles/projectile/beams.dm [hunk 1]
```diff
@@ -81,6 +85,26 @@
...
+	flag = "energy"
+	tracer_type = /obj/effect/projectile/tracer/laser/blue
+	muzzle_type = /obj/effect/projectile/muzzle/laser/blue
+	impact_type = /obj/effect/projectile/impact/laser/blue
+
+/obj/item/projectile/beam/laser/blue/triple
+	icon_state = "omnilaser"
+	damage = 6 // Does less damage due to cheaper energy cost.
+	eyeblur = 1
+	stamina = 5 // Bonus! It's function
```
Comment (ghost) [inline code/modules/projectiles/projectile/beams.dm:104]: 18+15 = 33 not 18

Comment (KathrinBailey) [reply code/modules/projectiles/projectile/beams.dm:104]: I forgot to finish the comment.
Does the 18 burn damage contribute to deciding when you get stam-crit?
This doesn't do 33 burn damage, it does 18 if all shots connect and 15 stamina.

-edit
oh yeah I know this has disabler icon state but it's just to make it a blue laser, it has the normal laser sound to not confuse people. It deals 6 burn, 5 stamina per shot, 18 burn and 15 stamina if they all connect.
It ain't obtainable outside of admin-spawn as well.

Comment (ghost) [reply code/modules/projectiles/projectile/beams.dm:104]: i know it isn't obtainable outside of admin spawn but omitting information in changelog is shitty. this is the same reason why we had instadeath for ipcs for half a year on emp because people don't know how to not lie through omission

Comment (Owai-Seek) [reply code/modules/projectiles/projectile/beams.dm:104]: Assuming they're lying on purpose instead of just missing something is a really shitty thing to do with something this big. Maintainers are supposed to review shit for a reason.

Comment (KathrinBailey) [reply code/modules/projectiles/projectile/beams.dm:104]: Urh, this isn't anywhere near that bad. I forgot, I added it in. I forgot to mention it in the word docx that I was recording changes in because I was half asleep whilst doing 70% of this PR.
Thanks for pointing it out

**Comment by Detective-Google** [comment]
those crafting recipes look like utter messes. I shouldn't need to break out the reading glasses to craft something. (ignoring all the other issues I have with this PR and it's targeting of a non existant improv weapon meta)

**Comment by KathrinBailey** [comment]
I'm not sure what "improvised weapon meta" you're referring to. You're going to have to elaborate and be less vague.
If you're having trouble reading the crafting menu interface you can use windows magnifier I guess? I'm not sure what you want me to do if you can't read the crafting menu.

**Comment by silicons** [comment]
outright:

i really wish people ran it with maintainers before doing things like adding a whole category of improvised weapons, even if the sprites looked cool, considering the issues we've had in the past with it.
i'll be back to properly comment later.

**Comment by lolman360** [comment]
the m90gl is a F2000 clone, not a P90, despite the name. the c20r is our p90esque weapon. so why are you making the f2000 shoot p90 rounds? and nerfing it, it's fine balancewise.

File: code/modules/projectiles/projectile/beams.dm (hunk 1 above)
Comment (lolman360) [reply code/modules/projectiles/projectile/beams.dm:104]: "never attribute to malice that which can be adequately explained by stupidity"
-gilette man (idk i dont use razors)

Comment (KathrinBailey) [reply code/modules/projectiles/projectile/beams.dm:104]: No off topic comments please

**Review by silicons — CHANGES_REQUESTED** [review]
I hate doing this but: this PR needs to be atomized:
EEDs are perfectly fine being in their own PR
Rebalances to what's currently nukeops weapons need to go into their own PR 
Overhauling the improvised weapons system needs to go into its own PR, and I'd appreciate two, one for the new laser stuff and one for rebalancing the current ones (and at this point I already do not like having improv guns have to use special improv ammo exclusively)
Misc stuff like handsaw is a whatever, same for loot drops, put them whereever you want

You really should have talked to one of us before doing this. I know this is tedious but I'm not going to sit here and review a major balance change to improvised weapon stuffed in the same PR as a balance change to nuclear operatives and feature additions.

**Comment by KathrinBailey** [comment]
Closing this because it will be atomised into smaller PRs.
With all the files changed it'll be a mess to take out parts to revert them.

Rebalancing of current improvised weapons is part 1.
Improvised pistol and laser will be part 2.
EEDs and Sumpter will be part 3.
